src/location.rs

```diff
@@ -6,77 +6,86 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use crate::xor_space::{Prefix, XorName};
-use std::fmt::{self, Debug, Formatter};
+use crate::{
+    error::{Result, RoutingError},
+    id::PublicId,
+    xor_space::{Prefix, XorName},
+};
 
-/// A `Location` is a source or destination of a message.
-///
-/// `Node` is single-node location (i.e. no verification of messages from
-/// additional sources needed). It's name is the `Authority::key` other
-/// locations require agreement by a quorum of `Elders`.
-#[derive(Serialize, Deserialize, PartialEq, PartialOrd, Eq, Ord, Clone, Copy, Hash)]
-pub enum Location {
-    /// A single section whose prefix matches the given name
-    Section(XorName),
-    /// A set of nodes with names sharing a common prefix - may span multiple `Section`s present in
-    /// the routing table or only a part of a `Section`
-    PrefixSection(Prefix<XorName>),
-    /// A single node
-    Node(XorName),
+/// Message source location.
+#[allow(clippy::large_enum_variant)]
+#[derive(Clone, Copy, Eq, PartialEq, Hash, Serialize, Deserialize, Debug)]
+pub enum SrcLocation {
```

Comment:
I wonder (just question) is `Src` and `Dst` alone perhaps not enough for these names?

Reply:
SrcLocation is more explicit and does not really add noise when used in code. I'd prefer we keep the full thing.

Reply:
Yeah, me too. It's bit more explicit without being that much more verbose.